# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/base/AbstractRange.cpp
# Commit: 7b5a689dc7fd
# Full Hash: 7b5a689dc7fd3105a920389b85ad84a2a5e5b715
# Author: Sean Feng <sefeng@mozilla.com>
# Date: 2024-03-23 09:29:17
# Regressor Bug: 1867058
# File Overlap Count: 1
# Description:
#   Bug 1867058 - Part 5: Update nsRange to create and update mCrossBoundaryRange when necessary r=smaug
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D195305
# ==============================================================================

diff -r 3cf108eb13a6 -r 7b5a689dc7fd dom/base/AbstractRange.cpp
--- a/dom/base/AbstractRange.cpp	Fri Mar 22 17:05:49 2024 +0000
+++ b/dom/base/AbstractRange.cpp	Fri Mar 22 17:05:50 2024 +0000
@@ -341,6 +341,8 @@
       // which they have been collapsed to one end, and it also may have a pair
       // of start and end which are the original value.
       aRange->DoSetRange(aEndBoundary, aEndBoundary, newEndRoot);
+      aRange->AsDynamicRange()->CreateOrUpdateCrossShadowBoundaryRangeIfNeeded(
+          aStartBoundary, aEndBoundary);
     }
     return NS_OK;
   }